backend/src/main/java/com/soen341/instagram/service/impl/FollowingService.java

```diff
@@ -0,0 +1,100 @@
+package com.soen341.instagram.service.impl;
+
+import java.util.Set;
+
+//Spring
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+//Project
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.exception.account.AccountNotFoundException;
+import com.soen341.instagram.exception.account.AlreadyFollowingException;
+import com.soen341.instagram.exception.account.SameAccountException;
+import com.soen341.instagram.model.Account;
+import com.soen341.instagram.utils.UserAccessor;
+
+@Service
+public class FollowingService {
+	
+	@Autowired
+	AccountRepository accountRepository;
+
+	public void follow(final String username)
+	{
+		Account following = UserAccessor.getCurrentAccount(accountRepository);
+		Account beingFollowed = getAccountToFollow(username);
+		
+		startFollowing(following, beingFollowed);
+		addFollower(following, beingFollowed);
+	}
+	
+	public void unfollow(final String username)
+	{
+		Account following = UserAccessor.getCurrentAccount(accountRepository);
+		Account beingFollowed = getAccountToFollow(username);
+		
+		stopFollowing(following, beingFollowed);
+		removeFollower(following, beingFollowed);
+	}
+	
+	private void startFollowing(final Account accountFollowing, final Account accountBeingFollowed)
+	{
+		if(accountFollowing.equals(accountBeingFollowed))
+		{
+			throw new SameAccountException("Accounts cannot follow themselves");
+		}
+		
+		Set<Account> followingSet = accountFollowing.getFollowing();
+		followingSet.add(accountBeingFollowed);
```

Comment:
I think it would be better to handle the `AlreadyFollowingException` here instead of  `addFollower`. In both case, if it fails in `startFollowing`, it would fail in `addFollower`.  The advantage of having it in `startFollowing` is that if it fails, you get to skip a method and skip querying the database thus increases performance maybe.